src/test.js

```diff
@@ -34,10 +34,15 @@ export default function Test( settings ) {
 	this.assertions = [];
 	this.semaphore = 0;
 	this.module = config.currentModule;
-	this.stack = sourceFromStacktrace( 3 );
 	this.steps = [];
 	this.timeout = undefined;
 
+	// evaluating error.stack can be expensive, so we should defer this if possible
+	var errorForStack = new Error();
+	this.getStack = function() {
```

Comment:
Have you considered using a getter here instead of adding a `.getStack()` method?

```js
// here
this._error = new Error();
this._stack = null;

// lower in the class body
get stack () {
  if (this._stack) { return this._stack; }

  return this._stack = extractStacktrace( errorForStack, 2 );
}
```

This would have the benefit of not actually changing the expected shape/api of the `Test` instance. Not sure we _really_ care about preserving the API, but it seems vaguely easier...

Reply:
👍 to this suggestion. This is how we've handled deferring stack prop access in other projects.

Reply:
That’s what I first thought of but I figured this was more compatible with old skool browsers. If that’s not an issue (because there’s some minimum version requirement or some compilation step that sorts this out) then it makes sense to implement this as a getter. Happy to take direction here.

Reply:
We do transpile with Babel, but QUnit also [currently supports](https://qunitjs.com/) the browsers [supported by jQuery 3.x](https://jquery.com/browser-support/), the oldest of which is IE9. 

[IE9 supports defining getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get#Browser_compatibility).